### PR TITLE
ENH: IntervalArray.min/max

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.4.0.rst [hunk 1]
```diff
@@ -218,6 +218,8 @@ Other enhancements
   ``USFederalHolidayCalendar``.  See also `Other API changes`_.
 - :meth:`.Rolling.var`, :meth:`.Expanding.var`, :meth:`.Rolling.std`, :meth:`.Expanding.std` now support `Numba <http://numba.pydata.org/>`_ execution with the ``engine`` keyword (:issue:`44461`)
 - :meth:`Series.info` has been added, for compatibility with :meth:`DataFrame.info` (:issue:`5167`)
+- Implemented :meth:`IntervalArray.min`, :meth:`IntervalArray.max`, as a result of which ``min`` and ``max`` now work for :class:`IntervalIndex`, :class:`Series` and :class:`DataFrame` with ``IntervalDtype`` (:issue:`44746`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.4.0.rst:221]: added in api reference?

File: pandas/core/arrays/interval.py [hunk 2]
```diff
@@ -790,6 +790,36 @@ def argsort(
             ascending=ascending, kind=kind, na_position=na_position, **kwargs
         )
 
+    def min(self, *, axis: int | None = None, skipna: bool = True):
+        if axis is not None and axis >= self.ndim:
+            raise ValueError(axis)
```
Comment (jreback) [inline pandas/core/arrays/interval.py:795]: case hit in tests?

**Review by jreback — CHANGES_REQUESTED** [review]
needs a rebase as well

File: doc/source/whatsnew/v1.4.0.rst (hunk 1 above)
Comment (jbrockmendel) [reply doc/source/whatsnew/v1.4.0.rst:221]: I don't think so.  The Index subclass already has it, and it doesn't look like we're doing that for the EA subclasses

File: pandas/core/arrays/interval.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/arrays/interval.py]: Updated with test, also fixes the mypy complaint on master

**Comment by jreback** [comment]
looks fine, can you merge master and ping on green

**Review by jreback — APPROVED** [review]

**Comment by jbrockmendel** [comment]
rebased + green-ish (same lint complaints as on master)

File: pandas/core/arrays/interval.py [hunk 3]
```diff
@@ -790,6 +790,36 @@ def argsort(
...
+        nv.validate_minmax_axis(axis, self.ndim)
+
+        if not len(self):
+            return self._na_value
+
+        mask = self.isna()
+        if mask.any():
+            if not skipna:
+                return self._na_value
+            return self[~mask].min()
```
Comment (jreback) [inline pandas/core/arrays/interval.py:803]: can u do

```suggestion
            self = self[~mask]
```

Comment (jbrockmendel) [reply pandas/core/arrays/interval.py:803]: sure.  follow-up OK?

Comment (jreback) [reply pandas/core/arrays/interval.py:803]: sure

**Review by jreback — APPROVED** [review]

**Pull request merged**